src/main/java/org/spongepowered/api/event/message/MessageSinkEvent.java

```diff
@@ -47,14 +49,14 @@
      *
      * @return The message sink the message in this event will be sent to
      */
-    MessageSink getSink();
+    Optional<MessageSink> getSink();
```

Comment:
Well IMO the same arguments apply here as well.

And for the possible argument "returns Optional.empty if not changed" IMO that only adds additional noise when using sinks and has no benefits at all. it even reduces the clarity of the method since one could asume the message shouldnt be send all of the sink is absent.
There is always a Sink the message will be broadcasted with and the magic value null/Optional.empty just hides this fact.
However if you keep the null stuff then you should at least explain in the javadocs what the values are supposed to express.